docs/paper/dev/guides/pdc.md

```diff
@@ -0,0 +1,132 @@
+ItemMeta meta = item.getItemMeta();
+item.getPersistentDataContainer().set(key, PersistentDataType.STRING, "I love Tacos!");
+item.setItemMeta(meta);
+```
+
+:::note NamespacedKeys
+
+It is considered good practice to store the `NamespacedKey`'s and reuse them.
+
+:::
+
+## Getting Data
+To get data from the PDC, you need to know the `NamespacedKey` and the `PersistentDataType` of the data.
+
+```java
+// Create a NamespacedKey
+NamespacedKey key = new NamespacedKey(pluginInstance, "example-key");
+
+ItemStack item = ...; // Retrieve the item from before
+// Get the data from the PDC
+PeristentDataContainer container = item.getItemMeta().getPersistentDataContainer();
+if (container.has(key, PersistentDataType.STRING)) {
+    String value = container.get(key, PersistentDataType.STRING);
+    // Do something with the value
+    player.sendMessage(Compontent.text(value));
+}
+```
+
+## Data Types
+
+The PDC supports a wide range of data types, such as:
+`Byte`, `Byte` `Array`, `Double`, `Float`, `Integer`, `Integer Array`, `Long`, `Long Array`, `Short`, `String` as well as `Tag Containers`.
+This means that you can store a wide range of data in the PDC. However, If you need a more complex data type, you can
+implement your own `PersistentDataType` and use that instead. Here is an example of how to do that for a UUID:
+
+```java
+public class UUIDDataType implements PersistentDataType<byte[], UUID> {
+     @Override
+     public Class<byte[]> getPrimitiveType() {
+         return byte[].class;
+     }
+
+     @Override
+     public Class<UUID> getComplexType() {
+         return UUID.class;
+     }
+
+     @Override
+     public byte[] toPrimitive(UUID complex, PersistentDataAdapterContext context) {
+         ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+         bb.putLong(complex.getMostSignificantBits());
+         bb.putLong(complex.getLeastSignificantBits());
+         return bb.array();
+     }
+
+     @Override
+     public UUID fromPrimitive(byte[] primitive, PersistentDataAdapterContext context) {
+         ByteBuffer bb = ByteBuffer.wrap(primitive);
+         long firstLong = bb.getLong();
+         long secondLong = bb.getLong();
+         return new UUID(firstLong, secondLong);
+     }
+ }
+```
+
+:::note
+
+In order to use your own `PersistentDataType`, you must pass an instance of it to the set/get methods.
+```java
+container.set(key, new UUIDDataType(), uuid);
+```
+
+:::
+
+## Storing on different objects
+
+- ##### Chunk
+    - `Chunk#getPersistentDataContainer()`
+- ##### World
+    - `World#getPersistentDataContainer()`
+- ##### Entity
+    - `Entity#getPersistentDataContainer()`
+- ##### TileState
+    - This is slightly more complicated, as you need to cast the block to something that extends `TileState`. 
+      This does not work for all blocks, only those that have a tile entity. F.E:
+      ```java
+        Block block = ...;
+        if (block instanceof Chest) {
```

Comment:
Consider the java version maybe can be good use `if (block instanceof Chest chest) {`? or like is is transversal maybe is better keep?